### PR TITLE
ZEPHYR-28114

### PR DESCRIPTION
**Pull request by pv-smartbear**

### CODE REVIEW

File: src/main/java/com/thed/zephyr/jenkins/reporter/ZeeReporter.java [hunk 1]
```diff
@@ -119,12 +119,12 @@ public boolean perform(final Run build, final TaskListener listener) throws IOEx
...
-                "[{ \"status\": \"$testsuite.testcase.failure\", \"statusExistPass\": false, \"statusString\": null, \"statusFailAttachment\": \"$testsuite.testcase.failure\", \"statusPassAttachment\": \"name: $testsuite.testcase:name \ntime: $testsuite.testcase:time\", \"packageName\": \"$testsuite:name\" , \"skipTestcaseNames\": \"\", \"testcase\" : {\"name\": \"$testsuite.testcase:name\"}, \"requirements\": [{\"id\": \"$testsuite.testcase.requirements.requirement\"}], \"attachments\": [{\"filePath\": \"$testsuite.testcase.attachments.attachment\"}]}]",//soapui
```
Comment (dinesh-belaguli) [inline src/main/java/com/thed/zephyr/jenkins/reporter/ZeeReporter.java]: These can be externalized is it? into a file system or DB (not part of this bug might be later)

File: src/main/java/com/thed/utils/ParserUtil.java [hunk 2]
```diff
@@ -227,6 +260,13 @@ Map deepCopy(Map originalMap) {
         return cloneMap;
     }
 
+    List deepyCopy(List originalList) {
+        Gson gson = new Gson();
+        String json = gson.toJson(originalList);
+        List cloneList = gson.fromJson(json, List.class);
```
Comment (dinesh-belaguli) [inline src/main/java/com/thed/utils/ParserUtil.java]: This deserializes into a new instance is it?

File: src/main/java/com/thed/utils/XMLLink.java [hunk 3]
```diff
@@ -59,6 +59,9 @@ public void attachLink(String[] linkArray, Type type) {
                 }
                 xmlLink.attachLink(Arrays.copyOfRange(linkArray, 1, linkArray.length), type);
             } else {
+                if(linkArray.length == 2) {
```
Comment (dinesh-belaguli) [inline src/main/java/com/thed/utils/XMLLink.java]: what does linkArray contains?

File: src/main/java/com/thed/zephyr/jenkins/reporter/ZeeReporter.java (hunk 1 above)
Comment (pv-smartbear) [reply src/main/java/com/thed/zephyr/jenkins/reporter/ZeeReporter.java]: Yes, there is a separate ticket for this to move to db.

File: src/main/java/com/thed/utils/ParserUtil.java (hunk 2 above)
Comment (pv-smartbear) [reply src/main/java/com/thed/utils/ParserUtil.java]: Yes, it creates deep copy without any relation to original.

File: src/main/java/com/thed/utils/XMLLink.java (hunk 3 above)
Comment (pv-smartbear) [reply src/main/java/com/thed/utils/XMLLink.java]: For a xml variable $testsuite.testcase.failure, link array will be ["testsuite", "testcase", "failure"]. These are links that we will traverse down while parsing.

**Pull request merged**